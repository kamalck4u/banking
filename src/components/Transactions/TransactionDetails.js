import React from 'react';
import { HeaderPanel, GenericErrorPanel } from '../Common/Panels';
import Navigations from '../Common/Navigations'
import { REFERENCE_NUMBER, DATE, DESCRIPTION, STATUS, AMOUNT, TRANSACTIONS_DETAILS } from '../Common/Constants';

const TransactionDetails = (props) => {
	const transactionDetails = props.location.state ? props.location.state.transaction : undefined;
	return (
		<React.Fragment>
			<Navigations back />
			{
				transactionDetails ? <HeaderPanel header={TRANSACTIONS_DETAILS} content={transactionDetails.from} /> : <HeaderPanel />
			}
			{
				transactionDetails ?
					<div className="table-responsive">
						<table className="table table-striped">
							<thead className="thead-dark">
								<tr>
									<th scope="col">{REFERENCE_NUMBER}</th>
									<th scope="col">{DATE}</th>
									<th scope="col">{DESCRIPTION}</th>
									<th scope="col">{AMOUNT}</th>
									<th scope="col">{STATUS}</th>
								</tr>
							</thead>
							<tbody>
								<tr key={transactionDetails.id}>
									<td>{transactionDetails.id}</td>
									<td>{transactionDetails.transaction_date}</td>
									<td>{transactionDetails.description}</td>
									<td>{transactionDetails.amount}</td>
									<td>{transactionDetails.transaction_processed === true ? 'Processed' : 'Pending'}</td>
								</tr>
							</tbody>
						</table>
					</div>
					: <GenericErrorPanel />
			}
		</React.Fragment>
	);
};
export default TransactionDetails;