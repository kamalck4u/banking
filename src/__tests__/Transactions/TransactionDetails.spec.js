import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import TransactionDetails from '../../components/Transactions/TransactionDetails';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const transaction = {
	'id': 1,
	'account_id': 1,
	'description': 'Test Description',
	'from': 'Savings Account - 342423455344',
	'transaction_date': '2019-06-08T03:37:28 -08:00',
	'transaction_processed': true,
	'amount': '$1,000.10'
};

let location = {
	state: {
		transaction: transaction
	}
};

describe('Transaction Details', () => {
	it('snapshot renders', () => {
		const TransactionListsComponent = renderer.create(<TransactionDetails location={location} />);
		expect(TransactionListsComponent.toJSON()).toMatchSnapshot();
	});
});

describe('Transaction Details', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<TransactionDetails location={location} />);
	});

	it('should show the transaction table headers', () => {
		expect(wrapper.find('th')).toHaveLength(5);
		expect(wrapper.find('th').getElements()[0].props.children).toEqual('Transaction#');
		expect(wrapper.find('th').getElements()[1].props.children).toEqual('Date');
		expect(wrapper.find('th').getElements()[2].props.children).toEqual('Description');
		expect(wrapper.find('th').getElements()[3].props.children).toEqual('Amount');
		expect(wrapper.find('th').getElements()[4].props.children).toEqual('Status');
	});

	it('should render the transaction details', () => {
		expect(wrapper.find('td')).toHaveLength(5);
		expect(wrapper.find('td').getElements()[0].props.children).toEqual(1);
		expect(wrapper.find('td').getElements()[1].props.children).toEqual('2019-06-08T03:37:28 -08:00');
		expect(wrapper.find('td').getElements()[2].props.children).toEqual('Test Description');
		expect(wrapper.find('td').getElements()[3].props.children).toEqual('$1,000.10');
		expect(wrapper.find('td').getElements()[4].props.children).toEqual('Processed');
	});

	it('should should transaction status as pending if not processed', () => {
		transaction.transaction_processed = false;
		wrapper = shallow(<TransactionDetails location={location} />);
		expect(wrapper.find('td').getElements()[4].props.children).toEqual('Pending');
	});

	it('should display account details on header panel', () => {
		expect(wrapper.find('HeaderPanel').props()).toEqual({ 'content': 'Savings Account - 342423455344', 'header': 'Transaction Details' });
	});

});
describe('Error', () => {
	it('should show generic error', () => {
		location = {};
		let wrapper = shallow(<TransactionDetails location={location} />);
		expect(wrapper.find('GenericErrorPanel')).toHaveLength(1);
	});
});

