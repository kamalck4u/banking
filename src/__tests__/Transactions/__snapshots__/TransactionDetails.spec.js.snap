// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Details snapshot renders 1`] = `
Array [
  <div
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <i
      aria-hidden="true"
      className="fa fa-chevron-left fa-2x nav-icon"
      onClick={[Function]}
    />
  </div>,
  <div
    className="jumbotron panel-header"
  >
    <h1
      className="display-4"
    >
      Transaction Details
    </h1>
    <p
      className="lead"
    >
      Savings Account - 342423455344
    </p>
    <hr
      className="my-4"
    />
    <p>
       
    </p>
  </div>,
  <div
    className="table-responsive"
  >
    <table
      className="table table-striped"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Transaction#
          </th>
          <th
            scope="col"
          >
            Date
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            Amount
          </th>
          <th
            scope="col"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            2019-06-08T03:37:28 -08:00
          </td>
          <td>
            Test Description
          </td>
          <td>
            $1,000.10
          </td>
          <td>
            Processed
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
